src/seq_analysis/orf.rs

```diff
@@ -126,9 +128,13 @@ where
     type Item = Orf;
 
     fn next(&mut self) -> Option<Orf> {
-        let mut result: Option<Orf> = None;
         let mut offset: usize;
 
+        // return any orfs already found
+        if !self.state.found.is_empty() {
+            return self.state.found.pop_front();
+        }
+
```

Comment:
If there are already found ORFs than haven't been reported yet, then this queue should be exhausted before moving back to scanning